src/main/java/net/dv8tion/jda/core/requests/restaction/MessageAction.java

```diff
@@ -480,13 +487,67 @@ public MessageAction addFile(final File file, final String name)
 
     /**
      * Clears all previously added files
+     * <br>And closes {@code FileInputStreams} generated by {@link #addFile(File, String)}.
      *
      * @return Updated MessageAction for chaining convenience
+     *
+     * @see    #clearFiles(Consumer)
+     * @see    #clearFiles(BiConsumer)
      */
     @CheckReturnValue
     public MessageAction clearFiles()
     {
         files.clear();
+        clearResources();
+        return this;
```

Comment:
Would by default close un-owned streams too (knowing ppl won't do it manually) and add an override with boolean parameter closeUnowned or something. same goes for other clearFiles() methods

Reply:
I don't want to close streams that were never used

Reply:
Yea but that would mean if a user uses channel.sendFile(someInputStream, "name") and never actually activates the restaction (or just calls clearFiles()), they would have to manually close the inputstream on their end. I think it would be nice if we would close any inputstream we get if they are either sent or the MessageAction is "canceled" (clearFiles).

Therefore i also suggested that rewrite of the docs (and probably add a big warning)

Reply:
If the user wants to close _all_ streams they can do so with `clearFiles(Consumer)`

Reply:
oh right... add that to some jdocs tho